servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/ScanWithPublisherTest.java

```diff
@@ -41,6 +44,7 @@
 import static org.hamcrest.Matchers.nullValue;
 
 public class ScanWithPublisherTest {
+
```

Comment:
can you some tests where the subscription and subscriber are on different threads, and verify the finalization is only invoked once?

Reply:
Added